error[E0080]: evaluation of constant value failed
  --> $DIR/infinite-recursion-const-fn.rs:4:5
   |
LL |     b()
   |     ^^^
   |     |
   |     reached the configured maximum number of stack frames
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
   |     inside `a`
...
LL |     a()
   |     ---
   |     |
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
   |     inside `b`
LL | }
LL | const ARR: [i32; a()] = [5; 6];
   |                  --- inside `ARR::{constant#0}`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0080`.
